test/distributed/algorithms/quantization/quantization_test.py

```diff
@@ -0,0 +1,108 @@
+import torch
+import os
+import torch.cuda
+import torch.distributed as dist
+import torch.distributed.algorithms.quantization.quantization as quant
+from torch.distributed.algorithms.quantization.quantization import DQuantType
+from torch.testing._internal.common_distributed import (
+    requires_gloo,
+    skip_if_lt_x_gpu,
+    requires_nccl,
+)
+from torch.testing._internal.distributed.distributed_test import (
+    DistributedTest, TestDistBackend
+)
+
+def _build_tensor(size, value=None, dtype=torch.float, device_id=None):
+    if value is None:
+        value = size
+    if device_id is None:
+        return torch.empty(size, size, size, dtype=dtype).fill_(value)
+    else:
+        return torch.empty(size, size, size, dtype=dtype).fill_(value).cuda(device_id)
+class DistQuantizationTests(TestDistBackend, DistributedTest._DistTestBase):
```

Comment:
what's the reason of importing these two base class for the collective tests? I think we can just base on `MultiProcessTestCase` similar to https://github.com/pytorch/pytorch/blob/master/test/distributed/algorithms/ddp_comm_hooks/test_ddp_hooks.py#L57 and https://github.com/pytorch/pytorch/blob/master/test/distributed/_sharded_tensor/test_sharded_tensor.py#L118 ?

Reply:
I see, but if you are trying to test different backends, shouldn't we skip some of the tests in certain backend? i.e. `all_to_all`  or `all_to_all_single` is not available in the gloo backend, we should skip it? we should ensure the collective only runs on the supported backend, we can ensure it like this https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/distributed/distributed_test.py#L860

Reply:
I thought `@requires_nccl()` would automatically do that. Can I use `sandcastle_skip_if` in this case?

Reply:
`requires_nccl` just ensure if `nccl` is not built together, the test get skipped, but it doesn't ensure we skip other backends properly, i.e. it might not skip the gloo backend for `all_to_all`. Yeah I think sandcastle_skip_if should be enough

Reply:
I see, okay then I will also add `sandcastle_skip_if`.